avro-serializer/src/main/java/io/confluent/kafka/serializers/subject/RecordNameStrategy.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.kafka.serializers.subject;
+
+import java.util.Map;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericContainer;
+import org.apache.kafka.common.errors.SerializationException;
+import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig;
+
+/**
+ * For any Avro record type that is published to Kafka, registers the schema
+ * in the registry under the fully-qualified record name (regardless of the
+ * topic). This strategy allows a topic to contain a mixture of different
+ * record types, since no intra-topic compatibility checking is performed.
+ * Instead, checks compatibility of any occurrences of the same record name
+ * across <em>all</em> topics.
+ */
+public class RecordNameStrategy implements SubjectNameStrategy {
+
+  @Override
+  public void configure(Map<String, ?> config) {
```

Comment:
I'm not seeing this being invoked after instance creation. Since this is a public API, users would possibly expect it to work if they use it. We should either invoke it or not extend Configurable in the interface.

Reply:
This is invoked [here](https://github.com/confluentinc/schema-registry/pull/680/files#diff-f0b851abbc98d83fb2946494a027f988R116), right?

I think it's useful to extend Configurable -- it's too hard w/ Java 7 to add it later when we need it.

Reply:
That's right. I missed it is using getConfiguredInstance. LGTM